source/Controlled.tsx

```diff
@@ -532,39 +540,64 @@ class ControlledBase extends Component<ControlledPropsWithDefaults, ControlledSt
   }
 
   /**
-   * Start tracking the Y-axis
+   * Start tracking the Y-axis but abort if non-scroll
+   * gesture is detected (like pinch-to-zoom)
    */
   handleTouchStart = (e: TouchEvent) => {
+    if (e.touches.length > 1) {
+      this.isScaling = true
+      return
+    }
+
     if (e.changedTouches.length === 1 && e.changedTouches[0]) {
       this.touchYStart = e.changedTouches[0].screenY
     }
   }
 
   /**
-   * Track how far we've moved on the Y-axis
-   * and unzoom if we detect a "swipe"
+   * If the window isn't browser zoomed,
+   * track how far we've moved on the Y-axis
+   * and unzoom if we detect a swipe
    */
   handleTouchMove = (e: TouchEvent) => {
-    if (this.touchYStart != null && e.changedTouches[0]) {
-      this.touchYEnd = e.changedTouches[0].screenY
-
-      const max = Math.max(this.touchYStart, this.touchYEnd)
-      const min = Math.min(this.touchYStart, this.touchYEnd)
-      const delta = Math.abs(max - min)
-      const threshold = 10
-
-      if (delta > threshold) {
-        this.touchYStart = undefined
-        this.touchYEnd = undefined
-        this.handleUnzoom()
+    if (!this.props.canSwipeToUnzoom) {
+      return
+    }
+
+    const browserScale = window.visualViewport?.scale ?? 1
+
+    if (!this.isScaling && browserScale <= 1 && this.touchYStart != null && e.changedTouches[0]) {
+      if (this.touchYStart != null && e.changedTouches[0]) {
```

Comment:
I see why this diff looks weird: it looks like your merge conflict resolution left this extra line of repeated code here.